pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -535,34 +491,34 @@ func (c *Controller) syncCheckStates(ctx context.Context, wl *kueue.Workload, ch
 					// to change to the "successfully provisioned" message after provisioning
 					updateCheckMessage(&checkState, apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Provisioned).Message)
 				}
-			case prAccepted:
-				if provisionedCond := apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Provisioned); provisionedCond != nil {
-					// propagate the ETA update from the provisioning request into the workload
-					updated = updateCheckMessage(&checkState, provisionedCond.Message) || updated
-					updated = updateCheckState(&checkState, kueue.CheckStatePending) || updated
-				}
+			case isAccepted(pr) && isProvisionedFalse(pr):
```

Comment:
I prefer the previous implementation, so that we don't hide the detail about what it means if the condition is not present.

Reply:
Done